src/controllerclientimpl.h

```diff
@@ -64,6 +64,10 @@ class ControllerClientImpl : public ControllerClient, public boost::enable_share
     virtual void DeleteDirectoryOnController_UTF8(const std::string& desturi);
     virtual void DeleteDirectoryOnController_UTF16(const std::wstring& desturi);
 
+    virtual void SaveBackup(std::vector<unsigned char>& vdata, bool config, bool media, double timeout);
+    virtual void RestoreBackup_UTF8(const std::string& filename_utf8, bool config, bool media);
```

Comment:
I think going forward, we should avoid forcing user to pass in filename, instead just pass in a stream or vector of bytes. You also don't need two functions in that case. And you have parity between SaveBackup and RestoreBackup.